### ISSUE TITLE
Duplicate endpoints identified

### ISSUE THREAD

**Issue by johnsimons** [Particular/ServiceControl#472]
When a user is not using heartbeats and the system contains multiple versions of NServiceBus core, endpoints are duplicated with incorrect nsb versions.

To replicate, run https://github.com/johnsimons/ReproDuplicateEndpoints

And then go to  http://localhost:33333/api/endpoints 
And you will see:
```json
[
{
"id": "a692078c-cf7a-dacb-02db-229f77b8ae65",
"name": "Publisher.ComplexSagaFindingLogic",
"host_display_name": "JOHNPC2",
"monitored": true,
"monitor_heartbeat": true,
"license_status": "unknown",
"is_sending_heartbeats": false
},
{
"id": "717ff783-0af5-637d-f194-499f2676c514",
"name": "Samples.ComplexSagaFindingLogic",
"host_display_name": "JOHNPC2",
"monitored": true,
"monitor_heartbeat": true,
"license_status": "valid",
"is_sending_heartbeats": false
},
{
"id": "0f07290e-6f06-a988-abc3-fe6e6d01ec3d",
"name": "Samples.ComplexSagaFindingLogic",
"host_display_name": "JOHNPC2",
"monitored": true,
"monitor_heartbeat": true,
"license_status": "valid",
"is_sending_heartbeats": false
},
{
"id": "50716f66-7914-35bd-abf2-4ef37fd8a207",
"name": "Subscriber.ComplexSagaFindingLogic",
"host_display_name": "JOHNPC2",
"monitored": true,
"monitor_heartbeat": true,
"license_status": "valid",
"is_sending_heartbeats": false
}
]
```

As you can see `Samples.ComplexSagaFindingLogic` is duplicated.
This is running v1.5.2

**Comment by gbiellem** [comment]
> This is running v1.5.3

@johnsimons  1.5.3? I think you mean 1.5.2

**Comment by johnsimons** [comment]
Fixed

**Comment by johnsimons** [comment]
@gbiellem feel like fixing this one ?

**Comment by gbiellem** [comment]
ok i'll take it

**Comment by gbiellem** [comment]
@johnsimons I tryied to repro against the 1.6 release branch and it looks like you've already fixed it - as you've commented here - https://github.com/Particular/ServiceControl/issues/472#issuecomment-108138292. Or am I missing something?

**Comment by johnsimons** [comment]
@gbiellem that comment was regarding your previous comment, where I added the wrong version

**Comment by johnsimons** [comment]
@gbiellem can u try it against 1.5.3 and then upgrade to 1.6 and see if the upgrade fixes it ?

**Comment by gbiellem** [comment]
@johnsimons Will do. Thanks

**Comment by gbiellem** [comment]
@johnsimons 

- reproed in 1.5.3 successfully with a fresh DB
- Upgraded to the current release branch and the duplicates  were removed 
- Rerun the ReproDuplicateEndpoints again 
- No Duplicates endpoints in  http://localhost:33333/api/endpoints 

So I believe you have already fixed it :)

**Comment by johnsimons** [comment]
Awesome @gbiellem

**Comment by johnsimons** [comment]
Fixed as part of all the 1.6 changes
